docs/UPGRADING.md

```diff
@@ -2,6 +2,270 @@
+    ... := pm.FindBoardWithFQBN(...)
+    ... := pm.ResolveFQBN(...)
+    ... := pm.Package(...)
+    ... := pm.GetInstalledPlatformRelease(...)
+    ... := pm.GetAllInstalledToolsReleases(...)
+    ... := pm.InstalledPlatformReleases(...)
+    ... := pm.InstalledBoards(...)
+    ... := pm.FindToolsRequiredFromPlatformRelease(...)
+    ... := pm.GetTool(...)
+    ... := pm.FindToolsRequiredForBoard(...)
+    ... := pm.FindToolDependency(...)
+    ... := pm.FindDiscoveryDependency(...)
+    ... := pm.FindMonitorDependency(...)
+}
+```
+
+Now we must obtain the `Explorer` object to access the same methods, moreover, we must call the `release` callback
+function once we complete the task:
+
+```go
+func DoStuff(pm *packagemanager.PackageManager, ...) {
+    pme, release := pm.NewExplorer()
+    defer release()
+
+    ... := pme.GetPackages()
+    ... := pme.GetCustomGlobalProperties()
+    ... := pme.FindPlatform(...)
+    ... := pme.FindPlatformRelease(...)
+    ... := pme.FindPlatformReleaseDependencies(...)
+    ... := pme.DownloadToolRelease(...)
+    ... := pme.DownloadPlatformRelease(...)
+    ... := pme.IdentifyBoard(...)
+    ... := pme.DownloadAndInstallPlatformUpgrades(...)
+    ... := pme.DownloadAndInstallPlatformAndTools(...)
+    ... := pme.InstallPlatform(...)
+    ... := pme.InstallPlatformInDirectory(...)
+    ... := pme.RunPostInstallScript(...)
+    ... := pme.IsManagedPlatformRelease(...)
+    ... := pme.UninstallPlatform(...)
+    ... := pme.InstallTool(...)
+    ... := pme.IsManagedToolRelease(...)
+    ... := pme.UninstallTool(...)
+    ... := pme.IsToolRequired(...)
+    ... := pme.LoadDiscoveries(...)
+    ... := pme.GetProfile(...)
+    ... := pme.GetEnvVarsForSpawnedProcess(...)
+    ... := pme.DiscoveryManager(...)
+    ... := pme.FindPlatformReleaseProvidingBoardsWithVidPid(...)
+    ... := pme.FindBoardsWithVidPid(...)
+    ... := pme.FindBoardsWithID(...)
+    ... := pme.FindBoardWithFQBN(...)
+    ... := pme.ResolveFQBN(...)
+    ... := pme.Package(...)
+    ... := pme.GetInstalledPlatformRelease(...)
+    ... := pme.GetAllInstalledToolsReleases(...)
+    ... := pme.InstalledPlatformReleases(...)
+    ... := pme.InstalledBoards(...)
+    ... := pme.FindToolsRequiredFromPlatformRelease(...)
+    ... := pme.GetTool(...)
+    ... := pme.FindToolsRequiredForBoard(...)
+    ... := pme.FindToolDependency(...)
+    ... := pme.FindDiscoveryDependency(...)
+    ... := pme.FindMonitorDependency(...)
+}
+```
+
+The `Explorer` object keeps a read-lock on the underlying `PackageManager` that must be released once the task is done
+by calling the `release` callback function. This ensures that no other task will change the status of the
+`PackageManager` while the current task is in progress.
+
+The `PackageManager.Clean()` method has been removed and replaced by the methods:
+
+- `PackageManager.NewBuilder() (*Builder, commit func())`
+- `Builder.BuildIntoExistingPackageManager(target *PackageManager)`
```

Comment:
Was it intentional to use a different parameter name here (`target` vs `old`)?

https://github.com/arduino/arduino-cli/blob/04e70c53f24c0b3738cbdfbf685aa8a3ca635217/arduino/cores/packagemanager/package_manager.go#L85

Reply:
I forget to change `old` to `target`, will do it.